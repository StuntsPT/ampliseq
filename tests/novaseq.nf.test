nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_novaseq"

    test("-profile test_novaseq") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                /* New syntax, is not satisfied
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_ampliseq_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
                */
                // Old syntax, provisionally used
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                path("$outputDir/dada2/ASV_table.tsv"),
                                path("$outputDir/dada2/DADA2_stats.tsv"),
                                path("$outputDir/dada2/DADA2_table.rds"),
                                path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/fastqc/S1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/S1_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/S2_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/S2_2_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/input/Samplesheet_novaseq.tsv")).match("input") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt")).match("multiqc") },
                { assert new File("$outputDir/summary_report/summary_report.html").exists() }
            )
        }
    }
}
