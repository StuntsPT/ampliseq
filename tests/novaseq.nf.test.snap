{
    "input": {
        "content": [
            "Samplesheet_novaseq.tsv:md5,bcca78fe24736f4ee4ea26fdbd47ee64"
        ],
        "timestamp": "2023-06-20T00:10:02+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,7b11038ff5eb3c795fc60ddf1cd17156",
            "ASV_table.tsv:md5,9735aeef0ecad9ea66e7bf2222a3ffb4",
            "DADA2_stats.tsv:md5,f509e598009556e0730b558022e75278",
            "DADA2_table.rds:md5,a475a9459dc32195e67e1e78ab048da4",
            "DADA2_table.tsv:md5,cb47a29726b9fa4671e022428d2825cd"
        ],
        "timestamp": "2025-03-25T00:10:02+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_fastqc.txt:md5,a99548124efd185a7d4e33245edcb81b",
            "multiqc_general_stats.txt:md5,92648df79061fa3b17fea104fe1b7641"
        ],
        "timestamp": "2023-06-20T00:10:02+0000"
    }
}
